packages/core/src/auth/auth.ts

```diff
@@ -135,6 +135,9 @@ export class Auth implements AuthService, ConnectionManager {
     /** Fired when a connection and its metadata has been completely deleted */
     public readonly onDidDeleteConnection = this.#onDidDeleteConnection.event
 
+    // User has attempted auth
+    public readonly onDidInitiateConnection = new vscode.EventEmitter<string | undefined>()
```

Comment:
This doesn't look right. This is adding a special-case without considering the semantics of this interface. This adds a new event but it's being controlled by other (amazonq) code . So listening to this from toolkit code would do nothing, which is very misleading.

Why can't this be captured by this existing `onDidChangeConnectionState` event, or one of the other (3!) existing "change" or "update" events?

Reply:
Also it doesn't look like there are actually any listeners to this event, so... what is its purpose?